sys/kern/tty.c

```diff
@@ -910,3 +911,31 @@ vnodeops_t tty_vnodeops = {
   .v_close = tty_vn_close,
   .v_getattr = tty_vn_getattr,
 };
+
+/* Controlling terminal pseudo-device (/dev/tty) */
+
+static int dev_tty_open(vnode_t *v, int mode, file_t *fp) {
```

Comment:
Why we don't use first argument in this function?

Reply:
Because we don't need it. What we need is the vnode of the controlling terminal.